File: shared/types/news.ts
export interface NewsArticle {
  id: number;
  headline: string;
  datetime: Date;
  url: string;
  source: string;
  category?: NewsCategory;
  image?: string;
  summary?: string;
}

export type NewsCategory =
  | "general"
  | "forex"
  | "crypto"
  | "merger"
  | "technology"
  | "business"
  | "top news";

File: backend/app/schemas/news.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

type NewsCategory = Literal["general", "forex", "merger", "technology", "business", "top news"] 

class NewsArticle(BaseModel):
    id: int
    headline: str
    datetime: datetime
    url: str
    source: str
    category: NewsCategory | None 
    image: str | None
    summary: str | None

